assets/js/blocks/featured-product/block.js

```diff
@@ -103,17 +106,34 @@ class FeaturedProduct extends Component {
 		const { productId } = this.props.attributes;
 		if ( ! productId ) {
 			// We've removed the selected product, or no product is selected yet.
-			this.setState( { product: false, loaded: true } );
+			this.setState( { product: false, loaded: true, error: false } );
 			return;
 		}
 		apiFetch( {
 			path: `/wc/blocks/products/${ productId }`,
 		} )
 			.then( ( product ) => {
-				this.setState( { product, loaded: true } );
+				this.setState( { product, loaded: true, error: false } );
 			} )
-			.catch( () => {
-				this.setState( { product: false, loaded: true } );
+			.catch( ( apiError ) => {
+				const error = {
+					retry: this.debouncedGetProduct,
+				};
+
+				if ( isObject( apiError ) ) {
+					error.message = (
+						<span dangerouslySetInnerHTML={
+							{
+								__html: escapeHTML( __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) ) + ': <br/><code>' + escapeHTML( apiError.message ) + '</code>',
```

Comment:
I know in a previous review I proposed this code, but looking at it again, I think we don't need `dangerouslySetInnerHTML` at all. This would work as well, right?
```JSX
error.message = (
	<span>
		{ __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) }
		<br />
		<code>{ escapeHTML( apiError.message ) }</code>
	</span>
);
```
Sorry for the back-and-forth, I didn't realize about it in the previous code review. :sweat: